### ISSUE TITLE
channel() with slack does not work any more

### ISSUE THREAD

**Issue by ASunc** [influxdata/kapacitor#2470]
When trying to send an alert to slack channel, it always goes to default channel of the slack webhook.
Ie. calling slack().channel('#xyz') won't have any effect.

I think this is because slack has deprecated legacy webhooks. Incoming webhooks don't support channel property.

**Comment by joshdurbin** [comment]
I've seen this issue too, with version `1.5.7`.

**Comment by docmerlin** [comment]
Thanks!

**Comment by erkexzcx** [comment]
Hi. Any updates on this issue?

**Comment by lesam** [comment]
Did you create your Incoming Webhook through the link in the docs? https://slack.com/services/new/incoming-webhook 

New-style Incoming Webhooks as part of slack apps are tied to the channel you created them on, but the legacy webhooks at that link should still work and allow configuring the correct channel.

We are working on supporting new-style webhooks and documenting the correct configuration.

**Comment by lesam** [comment]
The `token` argument below will be newly available configuration in Kapacitor 1.6:

```
	// The Slack webhook URL, can be obtained by adding Incoming Webhook integration.
 	// The Slack webhook URL.
 	// For new-style slack apps, use "https://slack.com/api/chat.postMessage".
 	// For legacy webhooks (e.g. created at https://slack.com/services/new/incoming-webhook), use the webhook link.
 	URL string `toml:"url" override:"url,redact"`
 	// The Slack OAuth token.
 	// For new-style slack apps, go to https://api.slack.com/apps -> your app -> 'OAuth & Permissions' to find
 	// the token. Ensure your app has 'chat:write' and 'chat:write.public' permissions.
 	// For legacy webhooks this can be left blank.
 	Token string `toml:"token" override:"token,redact"`
```
